src/main.ts

```diff
@@ -39,13 +39,13 @@ async function main() {
         await executeAzCliCommand("--version", true, execOptions);
         core.debug(`az cli version used:\n${output}`);
     
-        let creds = core.getInput('creds', { required: true });
-        let secrets = new SecretParser(creds, FormatType.JSON);
-        let servicePrincipalId = secrets.getSecret("$.clientId", false);
-        let servicePrincipalKey = secrets.getSecret("$.clientSecret", true);
-        let tenantId = secrets.getSecret("$.tenantId", false);
-        let subscriptionId = secrets.getSecret("$.subscriptionId", false);
-        let resourceManagerEndpointUrl = secrets.getSecret("$.resourceManagerEndpointUrl", false);
+        let creds = core.getInput('creds', { required: false });
+        let secrets = creds ? new SecretParser(creds, FormatType.JSON) : null;
+        let servicePrincipalId = core.getInput('creds_client_id', { required: false }) ? core.getInput('creds_client_id', { required: false}) : secrets.getSecret("$.clientId", false);
+        let servicePrincipalKey = core.getInput('creds_client_secret', { required: false }) ? core.getInput('creds_client_secret', { required: false}) : secrets.getSecret("$.clientSecret", false);
+        let tenantId = core.getInput('creds_tenant_id', { required: false }) ? core.getInput('creds_tenant_id', { required: false }) : secrets.getSecret("$.tenantId", false);
+        let subscriptionId = core.getInput('creds_subscription_id', { required: false }) ? core.getInput('creds_subscription_id', { required: false }) : secrets.getSecret("$.subscriptionId", false);
+        let resourceManagerEndpointUrl = creds ? secrets.getSecret("$.resourceManagerEndpointUrl", false) : "https://management.azure.com/";

```

Comment:
@anatbal resourceManagerEndpointUrl is different for sovereign clouds for example us gov-  "resourceManagerEndpointUrl": "https://management.usgovcloudapi.net/", we can't default it to "https://management.azure.com/". Ideally this should also be passed separately in that case.  @N-Usha ,@kmkumaran any workaround for this other than explicitly passing resourceManagerEndpoint.

Reply:
@BALAGA-GAYATRI, I changed the resourceManagerEndpointUrl to also be passed separately and fixed the default value. Let me know if that looks okay to you.